timer/timer.go

```diff
@@ -0,0 +1,152 @@
+package timer
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/jonboulle/clockwork"
+
+	"github.com/nivista/steady/internal/.gen/protos/messaging"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type (
+	// Timer is a recurring task that will invoke handlers when recurring tasks execute or are finished.
+	Timer interface {
+		// WithProgress should return a new timer with the given Progress.
+		WithProgress(pb *messaging.Progress) Timer
```

Comment:
this is important because when you first recieve a timer, you don't yet have the progress information. I saw three solutions:
1) this
2) have a wrapper around the timer that basically just holds the information needed to construct it, and then constructs it once it gets all the information (db query) - i didn't like this just because it was messy, the "timer wrapper that contains either an initialized timer or partial information to initialize one" 
3) call to Start takes a progress. This i felt breaks the Start Stop timer abstraction. If progress isn't "part" of a timer, why shouldn't you be able to call start multiple times with different progresses?

Reply:
As constructor! not interface.